File: vyger.Web/App_Start/FlashMessage.cs
namespace vyger.Web
{
    public enum FlashMessageType
    {
        Success,
        Info,
        Warning,
        Danger
    }

    public class FlashMessage
    {
        public FlashMessage(FlashMessageType type, string message)
        {
            Type = type;
            Message = message;
        }

        public FlashMessageType Type { get; private set; }

        public string Message { get; private set; }
    }
}

File: vyger-app/src/models/flash-message.ts
let id = 0;

export enum FlashMessageSeverity
{
    Information = 'Information',
    Success = 'Success',
    Danger = 'Danger'
}

export interface IFlashMessage
{
    severity?: FlashMessageSeverity;
    message?: string;
    growl?: boolean;
}

export class FlashMessage
{
    id: number;
    severity: FlashMessageSeverity;
    message: string;
    growl: boolean;
    seconds: number;

    constructor(properties?: IFlashMessage)
    {
        this.id = ++id;
        this.severity = FlashMessageSeverity.Information;
        this.seconds = 0;

        if (properties)
        {
            this.severity = properties.severity || this.severity;
            this.message = properties.message || this.message;
            this.growl = properties.growl || this.growl;
        }
    }
}
